client/my-sites/themes/controller/index.node.js

```diff
@@ -42,27 +40,14 @@ export function fetchThemeDetailsData( context, next ) {
 	}
 
 	const themeSlug = context.params.slug;
-	const theme = themeDetailsCache.get( themeSlug );
+	const theme = getThemeDetails( context.store.getState(), themeSlug );
 
-	if ( theme ) {
+	if ( Object.keys( theme ).length ) {
```

Comment:
To move forward, I'd suggest merging this PR so we get at least the benefit

> This should also pick up re-hydrated `themeDetail` from `initialReduxState` on the client-side, which is a nice side effect.

and some more easily understandable code. Persisting theme details API cache across requests can be the subject of a separate PR.

We should however keep in mind that having `long_description`s for ~300 themes in `initialReduxState` would blow up our initial HTML really badly!